### ISSUE TITLE
Dropout type error?

### ISSUE THREAD

**Issue by jmhessel** [fchollet/keras#1700]
I am attempting to implement [Deep Averaging Networks](https://github.com/miyyer/dan) in keras. The model applies dropout to words in a sequence prior to embedding them. So, the thing I want to do is basically (assuming that I have padded sequences of length maxlen, and I have max_features tokens).

    model = Sequential()
    model.add(Dropout(.3, input_shape = [maxlen]))
    model.add(Embedding(max_features, embedding_dim))
    model.add(LSTM(128))
    model.compile(loss='binary_crossentropy', optimizer='adagrad', class_mode="binary")

This relates to another issue I submitted tonight (#1698) but for the purposes of this issue -- it seems that dropout kills the type of the input (I think it makes it a float). Is it worth supporting dropout over ints? I could see an argument that this is such a weird case that the API shouldn't be broken for this, but it would be nice to index into Embedding layers based on dropped out ints. I don't really see why Dropout should change the typing.

**Comment by jmhessel** [comment]
Conversely, dropout probably does more than simply dropping units out -- i.e. it has to up-multiply weights for the testing pass for the following layer. I am very sure, then, that this is not dropout's intended use. Perhaps it would be better to simply do batch training and implement this dropout mechanism on my own. Food for thought, nonetheless.

**Comment by jmhessel** [comment]
Ehh, on second thought, I don't really think this is an issue worth solving.

**Comment by alyato** [comment]
@jmhessel  i check the Embedding layer  and it says

> This layer can only be used as the first layer in a model.

But The Dropout is the first layer in your code.

> model.add(Dropout(.3, input_shape = [maxlen]))
model.add(Embedding(max_features, embedding_dim))
